llvm/tools/llvm-no-spir-kernel/llvm-no-spir-kernel.cpp

```diff
@@ -44,15 +51,21 @@ int main(int argc, char **argv) {
 
   // Use lazy loading, since we only care about function calling convention
   SMDiagnostic Err;
+  const char *ProgramName = llvm::sys::path::filename(argv[0]).data();
   std::unique_ptr<Module> M = getLazyIRFileModule(InputFilename, Err, Context);
 
   if (!M.get()) {
-    Err.print(argv[0], errs());
-    return 1;
+    Err.print(ProgramName, errs());
+    return -1;
   }
 
   for (auto &F : *M) {
     if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
+      std::string SPIRKernelMsg =
+          "Unexpected SPIR kernel occurrence: " + demangle(F.getName().str());
```

Comment:
Do we really need to demangle function name?

Reply:
It boils down to the user preferences. An example would be `typeinfo name for run_kernel_add(cl::sycl::queue&, float, float, float&)::$_0::operator()(cl::sycl::h andler&) const::SimpleAdd` vs `_ZTSZZ14run_kernel_addRN2cl4sycl5queueEffRfENK3$_0clERNS0_7handlerEE9SimpleAdd`.

As compiler developers, we may have no strong preference/be adapt at eye-demangling. Perhaps @domiyan would be the best judge as to the SYCL FPGA users' preferences.

Reply:
@domiyan, could you please resolve this conversation if it's fine with you?